csrc/scheduler/tools/loop_domain_scheduler.cpp

```diff
@@ -498,99 +502,125 @@ void scheduleLoopDomainsLike(
-      for (const auto loop_id : tv->getLoopDomain()) {
-        if (output_g->has(loop_id)) {
-          output_ids.push_back(loop_id);
+      std::vector<IterDomain*> output_ids;
+      output_ids.reserve(transform->outputs().size());
+      for (const auto& output_g : output_groups) {
+        for (const auto loop_id : tv->getLoopDomain()) {
+          if (output_g->has(loop_id)) {
+            output_ids.push_back(loop_id);
+          }
         }
       }
-    }
 
-    // If all of the inputs are found, the tranform expr is replayed as
-    // a forward op.
-    Direction replay_dir_tv = Direction::Undefined;
-    if (replay_dir != Direction::Backward &&
-        input_ids.size() == transform->inputs().size()) {
-      replay_dir_tv = Direction::Forward;
-    } else if (
-        replay_dir != Direction::Forward &&
-        output_ids.size() == transform->outputs().size()) {
-      replay_dir_tv = Direction::Backward;
-    } else {
-      // Replay not possible since none of inputs nor outputs are connected with
-      // the transform
-      continue;
-    }
+      // If all of the inputs are found, the tranform expr is replayed as
+      // a forward op.
+      Direction replay_dir_tv = Direction::Undefined;
+      if (replay_dir != Direction::Backward &&
+          input_ids.size() == transform->inputs().size()) {
+        replay_dir_tv = Direction::Forward;
+      } else if (
+          replay_dir != Direction::Forward &&
+          output_ids.size() == transform->outputs().size()) {
+        replay_dir_tv = Direction::Backward;
+      } else {
+        // Replay not possible since none of inputs nor outputs are connected
+        // with the transform
+        continue;
+      }
 
-    // When the direction is forward, the TensorView transform
-    // APIs, e.g., TensorView::split, can be used, which doesn't need
-    // to use TensorView::setLoopDomain. This is important as
-    // setLoopDomain may result in losing extra IDs added by prior
-    // scheduleLoopDomain calls, which was indeed the case with the
-    // Llama 3 RoPE backward (see also
-    // https://github.com/NVIDIA/Fuser/issues/3571).
-    if (replay_dir_tv == Direction::Forward) {
-      ReplayForwardTransformOnLoopDomain::replayAs(tv, input_ids, transform);
-      continue;
-    }
+      // When the direction is forward, the TensorView transform
+      // APIs, e.g., TensorView::split, can be used, which doesn't need
+      // to use TensorView::setLoopDomain. This is important as
+      // setLoopDomain may result in losing extra IDs added by prior
+      // scheduleLoopDomain calls, which was indeed the case with the
+      // Llama 3 RoPE backward (see also
+      // https://github.com/NVIDIA/Fuser/issues/3571).
+      if (replay_dir_tv == Direction::Forward) {
+        auto replayed_expr = ReplayForwardTransformOnLoopDomain::replayAs(
+            tv, input_ids, transform);
+        NVF_ERROR_EQ(replayed_expr->outputs().size(), output_groups.size());
```

Comment:
```suggestion
        NVF_ERROR_EQ((int64_t)replayed_expr->outputs().size(), (int64_t)output_groups.size());
```
**[Suggested by AI]** The comparison inside the NVF_ERROR_EQ macro failed to compile because the two arguments were of different (size‐related) types.  
The patch casts both operands to int64_t:

• Before: compared outputs().size() and output_groups.size() directly → type mismatch caused build error.  
• After: both values are explicitly cast to int64_t, giving them the same type and allowing the macro to compile.

So the fix is simply adding the static casts to resolve the build-time type mismatch.